.github/workflows/ci.yml

```
@@ -0,0 +1,237 @@
+      - run: pip install ".[test]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto
+  test_pyargs:
+    name: test --pyargs
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+        exclude:
+          - os: macos-latest
+            python: '3.10'
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-pyargs-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install ".[test]" pytest-xdist
+      - run: pip freeze
+      - run: pytest -n auto ./docs --pyargs romanisim
+  test_older_numpy:
+    name: test Numpy ${{ matrix.numpy }} (Python ${{ matrix.python }})
+    needs: [ test ]
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        python: [ 3.8, 3.9, '3.10' ]
+        numpy: [ '1.18.*', '1.19.*', '1.20.*', '1.21.*', '1.22.*' ]
+        exclude:
+          - python: '3.10'
+            numpy: '1.18.*'
+          - python: '3.10'
+            numpy: '1.19.*'
+          - python: '3.10'
+            numpy: '1.20.*'
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-numpy${{ matrix.numpy }}-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install -e ".[test]" pytest-xdist
+      - run: pip install numpy==${{ matrix.numpy }}
+      - run: pip freeze
+      - run: pytest -n auto
+  test_with_coverage:
+    name: test with coverage
+    needs: [ test, test_alldeps, test_devdeps, test_pyargs, test_older_numpy ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.0.2
+      - uses: actions/setup-python@v4.1.0
+        id: python
+        with:
+          python-version: 3.9
+      - uses: actions/cache@v3.0.5
+        with:
+          path: ${{ env.pythonLocation }}
+          key: test-coverage-${{ runner.os }}-${{ env.pythonLocation }}-${{ hashFiles('**/pyproject.toml', '**/setup.*') }}
+      - run: pip install ".[test]" pytest-xdist pytest-cov
+      - run: pip freeze
+      - run: pytest -n auto --pyargs romanisim --cov-report xml --cov
+      - run: coverage report -m
+      - uses: codecov/codecov-action@master
```

```suggestion
      - uses: codecov/codecov-action@v3
```

This action uses a main branch not a master branch, which is why the CI is failing right now. It is better to pin a major release version, the latest appears to be `v3`.

---

Yes this is my fault, sorry, I introduced a fix for this in https://github.com/spacetelescope/romancal/pull/592

---

Changed.

---

Done